### DISCUSSION TITLE
[ 命名建議 ] LeetCode 739. Daily Temperatures

### DISCUSSION THREAD

**Discussion by LPenny-github** [EngTW/English-for-Programmers#119] [Q&A] [answered]
LeetCode:
https://leetcode.com/problems/daily-temperatures/

線上教材:
https://blog.techbridge.cc/2019/10/26/leetcode-pattern-next-greater-element/

---

我私自改了方法的名字，從 `DailyTemperatures` 變成 `GetWaitingDayUntilWarmerTemperature`，因為想練習 create, set, get 這幾個動詞的用法。雖然看了之前大大開的 issue (https://github.com/EngTW/English-for-Programmers/issues/28) 但我用起來還是有點兩光 😅😅😅 再麻煩 @twy30 給予命名建議 orz 感激不盡

```csharp
using System;
using System.Collections.Generic;

public class Solution
{
    public int[] GetWaitingDayUntilWarmerTemperature(int[] T)
    {
        int[] temperatures = T;
        Stack<int> nextWarmerDayStack = new Stack<int>();
        int[] waitingDayArray = new int[temperatures.Length];

        Array.Fill(waitingDayArray, 0);

        for (int i = 0; i < temperatures.Length; ++i)
        {
            while (nextWarmerDayStack.Count > 0 && 
                    temperatures[nextWarmerDayStack.Peek()] < temperatures[i])
            {
                int index = nextWarmerDayStack.Peek();
                waitingDayArray[nextWarmerDayStack.Pop()] = i - index;
            }
            nextWarmerDayStack.Push(i);
        }
        return waitingDayArray;
    }
}
```

**Comment by twy30** [comment, marked as answer]
@LPenny-github 

你好 😊

---

```C#
    public int[] GetWaitingDayUntilWarmerTemperature(int[] T)
```

我覺得 OK

---

```C#
        Stack<int> nextWarmerDayStack = new Stack<int>();
```

`nextWarmerDayStack` 的字面意思是「 *之後的* 、更暖的、 "天" 、 stack 」。

但從這個變數的實際用途來看，它是儲存「 *之前的* / 已檢視過的、 `temperatures` 的索引、 stack」。

或許叫它 `pastDayIndexes` 會更貼切 🤔

---

```C#
        int[] waitingDayArray = new int[temperatures.Length];
```

可以考慮

* `waitingDays`
* `output`

---

以下與命名無關，這個 `Array.File()` 是可以省掉的。

```C#
        Array.Fill(waitingDayArray, 0);
```

在 https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/default-values 指出： "built-in integral numeric type" (內建的整數數值類別； https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/integral-numeric-types ) 的預設值都是 0 。

**Reply by LPenny-github** [reply]
> 以下與命名無關，這個 Array.File() 是可以省掉的。

這邊可能是 `Array.Fill()` 😊？

謝謝 @twy30 的指導 orz

**Reply by twy30** [reply]
> > 以下與命名無關，這個 Array.File() 是可以省掉的。
> 
> 這邊可能是 `Array.Fill()` 😊？

你說的對，我的筆誤 😅 感謝指正 😊
